test/tla/TestSets.tla

```diff
@@ -0,0 +1,252 @@
+TestPowerset26 ==
+    { {}, { 2 }, { 6 }, { 2, 6 } } = SUBSET Set26
+
+\* TODO: FIX IT!
+\* https://github.com/informalsystems/apalache/issues/1103
+TestPowersetSubsetEq ==
+    (SUBSET Set26) \subseteq SUBSET Set26
+
+\* TODO: FIX IT!
+\* https://github.com/informalsystems/apalache/issues/1339
+TestExistsEmptyInPowerset ==
+    \E S \in SUBSET Set263:
+        S = {}
+
+\* TODO: FIX IT!
+\* https://github.com/informalsystems/apalache/issues/1339
+TestExists15InPowerset ==
+    \E S \in SUBSET Set263:
+        S = { 2, 6 }
+
+TestForallInPowerset ==
+    \A S \in SUBSET Set1357:
+        6 \notin S
+
+\* TODO: FIX IT!
+\* https://github.com/informalsystems/apalache/issues/1360
+TestPowersetCardinality ==
+    Cardinality(SUBSET Set263) = 8
+
+TestUnion ==
+    UNION { Set26, Set1357, SetEmpty } = Set123567
+
+TestUnionEmpty ==
+    LET \* @type: Set(Set(Int));
+        Empty == {}
+    IN
+    UNION Empty = {}
+
+TestForall ==
+    \A i \in Set1357:
+        i = 1 \/ i = 3 \/ i = 5 \/ i = 7
+
+TestExists ==
+    \E i \in Set1357:
+        i = 3
+
+TestChooseMin ==
+    LET min == CHOOSE i \in Set1357:
+        \A j \in Set1357: i <= j
+    IN
+    min = 1
+
+TestChooseSet ==
+    LET S ==
+        CHOOSE T \in { SetEmpty, Set263, Set1357 }:
+            6 \in T
+    IN
+    S = Set263
+
+TestChooseInSingleton ==
+    LET x == CHOOSE i \in { 3 }: TRUE IN
+    x = 3
+
+TestChooseEmpty ==
+    LET x == CHOOSE i \in SetEmpty: TRUE IN
+    x <= 0 \/ x > 0
+
+TestForallSet ==
+    \A S \in { SetEmpty, Set26, Set1357 }:
+        10 \notin S
+
+TestExistsSet ==
+    \E S \in { SetEmpty, Set26, Set1357 }:
+        5 \in S
+
+TestIsFiniteSet ==
+    /\ IsFiniteSet(SetEmpty)
+    /\ IsFiniteSet(Set26)
+    /\ IsFiniteSet(Set1357)
+
+TestIsFiniteSetOnInfinite ==
+    /\ ~IsFiniteSet(Int)
+    /\ ~IsFiniteSet(Nat)
+
+AllTests ==
+    /\ TestSet123567eq123567
+    /\ TestSet123567ne1357
+    /\ TestSet123567neEmpty
+    /\ TestSet263ne26
+    /\ TestSet263union1357
+    /\ TestSet263minus1357
+    /\ TestSet263intersect1357
+    /\ TestSetTimes
+    /\ TestSetTimesCardinality
+    /\ TestCardinality123567
+    /\ TestCardinality1357
+    /\ TestCardinalityEmpty
+    /\ TestCardinalitySets
+    /\ TestIn
+    /\ TestNotIn
+    /\ TestSubseteq
+    /\ TestNotSubseteq
+    /\ TestSetMap
+    /\ TestSetMapSets
+    /\ TestForallInPowerset
+    /\ TestUnion
+    /\ TestUnionEmpty
+    /\ TestForall
+    /\ TestExists
+    /\ TestForallSet
+    /\ TestExistsSet
+    /\ TestChooseMin
+    /\ TestChooseInSingleton
+    /\ TestChooseEmpty
+    /\ TestChooseSet
+    /\ TestIsFiniteSet
+    \* IGNORE UNTIL FIXED: /\ TestIsFiniteSetOnInfinite
+    \* IGNORE UNTIL FIXED: /\ TestPowersetCardinality
+    \* IGNORE UNTIL FIXED: /\ TestPowersetEmpty
+    \* IGNORE UNTIL FIXED: /\ TestPowerset26
+    \* IGNORE UNTIL FIXED: /\ TestPowersetSubsetEq
+    \* IGNORE UNTIL FIXED: /\ TestExistsEmptyInPowerset
+    \* IGNORE UNTIL FIXED: /\ TestExists15InPowerset
```

Comment:
Is there a way to make sure we do not to miss uncommenting these once the issues are fixed?

Reply:
There are issues for all of these tests, so we will uncomment them once the issues are fixed